chapters/entry_execution.md

```diff
@@ -0,0 +1,103 @@
+# Entry Point, Execution Model, and Execution Mode
+
+The start of every SPIR-V module will contain the `OpEntryPoint` and `OpExecutionMode`/`OpExecutionModeId` instructions. This chapter is to take a more indepth look into them and edge cases tool developrs will need to watch out for.
+
+> Note: It is very easy to mix up the term `model` and `mode` if not reading carefully
+
+# Multiple Entry Points
+
+A SPIR-V module can have more than one Entry Point. This allows the possibility of sharing SPIR-V what otherwise would have to be a second SPIR-V module and possible code duplication.
+
+Each `OpEntryPoint` has the `Execution Model` and `Name` value with an important phrase in the spec
+
+> A module must not have two OpEntryPoint instructions with the same Execution Model and the same Name string.
+
+This means the following is legal SPIR-V:
+
+```swift
+OpEntryPoint GLCompute %main "name_a"
+OpEntryPoint GLCompute %main "name_b"
+```
+
+or
+
+```swift
+OpEntryPoint Vertex %main "name" %interface
+OpEntryPoint CallableKHR %main "name"
+```
+
+But the following will fail:
+
+```swift
+OpEntryPoint GLCompute %main_a "name_c"
+OpEntryPoint GLCompute %main_b "name_c"
+
+// error: 2 Entry points cannot share the same name and ExecutionMode.
+```
+
+This means any tool will have to be mindful that an **entry point is not unique to single value**. Tool consuming SPIR-V must be aware that two entry points can share the same `Execution Model` or `Name` operand of multiple `OpEntryPoint`.
```

Comment:
I'm not clear on what the emphasized phrase is supposed to mean exactly, because I'm not sure what "value" refers to.

Do you mean

"This means a simple one-to-one relationship is not guaranteed between:
- entry points and their associated functions
- an entry point name and its associated functions
"

?

For wording on the second sentence, I suggest changing:

> Tool consuming SPIR-V must be aware that two entry points can share the same `Execution Model` or `Name` operand of multiple `OpEntryPoint`.

to 

> Tools consuming SPIR-V must be aware that two entry points can use the same value for the  `Execution Model` or `Name` operands on their `OpEntryPoint` instructions, but not both at the same time.

Reply:
agree, `value` is not a good word choice here, changed the emphasized part to 

> entry point is not unique to single Execution Model and Name